Make YubiKey polling interval configurable

rafaelx opened the pull request:
This is an untested attempt to make the Yubi Key polling interval configurable via command line argument "-y".
This PR is related to the suggestion in #882 and has not been tested due to the lack of a Yubi Key.

1. Should the command line argument be "-y, --yubi_polling_interval" or "-p, --polling"?
2. Should an environment variable be used rather than a command line option?

ronys commented:
How's the first commit related? Was it added to this branch by mistake?

I'd start by having this only as a command-line argument, no env var, if for no other reason than it's more clearly documented (via --help).
For consistency, I'd call it '--yubi-poll-interval' ('-', not '_'). If possible, I'd also disable the "short" option ("-y"), as it's not going to be used very often.

dskrvk commented on src/ui/wxWidgets/SafeCombinationSetupDlg.cpp:
```diff
@@ -84,7 +84,9 @@ SafeCombinationSetupDlg::SafeCombinationSetupDlg(wxWindow *parent, wxWindowID id
 #ifndef NO_YUBI
   SetupMixin(FindWindow(ID_YUBIBTN), FindWindow(ID_YUBISTATUS));
   m_pollingTimer = new wxTimer(this, POLLING_TIMER_ID);
```
Do we even need to initialize the timer if polling isn't enabled?

rafaelx replied:
Changed as suggested.

dskrvk commented on src/ui/wxWidgets/YubiMixin.cpp:
```diff
@@ -81,6 +84,25 @@ void YubiMixin::HandlePollingTimer()
+  }
+  else {
+    s_pollingInterval = value;
+    s_pollingEnabled = true;
```
Nit: if the only case when polling is disabled is the < 0 case, why not move this repeating line out of the if branches? E.g.
```c++
  s_pollingEnabled = true;
  if (value <= 0) {
    s_pollingInterval = YubiMixin::POLLING_INTERVAL;
    s_pollingEnabled = false;
  }
```

rafaelx replied:
Changed as suggested.

dskrvk commented on src/ui/wxWidgets/YubiMixin.h:
```diff
@@ -25,7 +25,7 @@ class wxWindow;
 {
  public:
-  enum {POLLING_INTERVAL = 500}; // mSec
+  enum {POLLING_INTERVAL_MIN = 100, POLLING_INTERVAL = 500, POLLING_INTERVAL_MAX = 60000}; // mSec
```
Nit: maybe rename POLLING_INTERVAL to POLLING_INTERVAL_DEFAULT?

rafaelx replied:
Changed as suggested.

rafaelx commented:
> How's the first commit related? Was it added to this branch by mistake?
> 
> I'd start by having this only as a command-line argument, no env var, if for no other reason than it's more clearly documented (via --help). For consistency, I'd call it '--yubi-poll-interval' ('-', not '_'). If possible, I'd also disable the "short" option ("-y"), as it's not going to be used very often.

I accidentally didn't branch from Master. Corrected via a forced push to a new branch.
Removed the short option and renamed the long option as suggested.

ronys commented on src/ui/wxWidgets/DbSelectionPanel.cpp:
```diff
@@ -124,7 +126,10 @@ DbSelectionPanel::DbSelectionPanel(wxWindow* parent,
 DbSelectionPanel::~DbSelectionPanel()
 {
-  delete m_pollingTimer;
+  if (m_pollingTimer != nullptr) {
```
The if statement isn't needed, since the language guarantees that delete-ing a nullptr is safe.
Also, I don't think that setting m_pollingTimer to nullptr is needed, since when the destructor function is done, nothing should be able to access this member.
(This comment applies to all destructors)
